### DISCUSSION TITLE
Increase the scope of `out var`

### DISCUSSION THREAD

**Discussion by rexcfnghk** [dotnet/csharplang#5448] [Language Ideas] [answered]
(Hello! It's my first time suggesting a language idea so please excuse me if this is written poorly)

I would like to suggest increasing the scope of `out var` variables such that this refactoring would work:

```c#
// Original code
if (someBoolean && someDictionary.TryGetValue("key", out var myValue))
{
    // Compiles
    Console.WriteLine(myValue);
}

// Refactored code
var myCondition = someBoolean && someDictionary.TryGetValue("key", out var myValue);
if (myCondition)
{
    // Compile error CS0165: Use of unassigned local variable
    Console.WriteLine(myValue);
}
```
Is it possible to increase the scope of `out var` variables (i.e. `myValue` in my example) so that the code would compile?

**Comment by khm1600** [comment, marked as answer]
It's definite assignment that's complaining here, not scoping: 

```cs
var myCondition = someDictionary.TryGetValue("key", out var myValue);
if (myCondition)
{
    // Compiles
    Console.WriteLine(myValue);
}
```

**Reply by RikkiGibson** [reply]
This is correct. In the "working" version we have an expression which only definitely assigns a value to 'myValue' when the value of the expression is 'true'. This conditional information about the definite assignment state is lost when the expression is assigned to a local and then read.

**Comment by bernd5** [comment]
Same as:

```csharp
using System;

var someDictionary = new System.Collections.Generic.Dictionary<String, int>();
var someBoolean = true;

int myValue;
var myCondition = someBoolean && someDictionary.TryGetValue("key", out myValue);
if (myCondition)
{
    // Compile error CS0165: Use of unassigned local variable
    Console.WriteLine(myValue);
}
```
